icu4j/tools/build/src/main/java/com/ibm/icu/dev/tool/errorprone/ErrorProneReport.java

```diff
@@ -266,4 +268,107 @@ private static void outSummary(HtmlUtils hu, Map<String, List<ErrorProneEntry>>
             hu.closeTag("p").nl();
         }
     }
+
+    private static void genReportTsv(Map<String, List<ErrorProneEntry>> errors,
+            String outDir) throws IOException {
+        Path outFileName = Paths.get(outDir, TSV_REPORT_FILE);
+        System.out.println("Report generated: " + outFileName);
+        try (PrintStream wrt = new PrintStream(outFileName.toString(), StandardCharsets.UTF_8)) {
+            wrt.println("Issue type"
+                    + "\tCount"
+                    + "\tSeverity"
+                    + "\tURL"
+            );
+            for (Map.Entry<String, List<ErrorProneEntry>> e : errors.entrySet()) {
+                String errorType = e.getKey();
+                List<ErrorProneEntry> errorsOfType = e.getValue();
+                if (errorsOfType.isEmpty()) {
+                    continue;
+                }
+
+                String tags = ErrorProneUtils.getTags(errorType);
+                if (tags != null) {
+                    errorType += " " + tags;
+                }
+                ErrorProneEntry firstEntry = errorsOfType.get(0);
+                String errorSeverity = firstEntry.severity;
+
+                wrt.println(errorType
+                        + "\t" + errorsOfType.size()
+                        + "\t" + errorSeverity
+                        + "\t" + firstEntry.url
+                );
+            }
+        }
+    }
+
+
+    private static void genReportMd(Map<String, List<ErrorProneEntry>> errors,
+            String outDir) throws IOException {
+        Path outFileName = Paths.get(outDir, MD_REPORT_FILE);
+        System.out.println("Report generated: " + outFileName);
+        try (PrintStream wrt = new PrintStream(outFileName.toString(), StandardCharsets.UTF_8)) {
+            wrt.println("| Issue type | Severity | Location | Message |");
+            wrt.println("| ---------- | -------- | -------- | ------- |");
+            for (Map.Entry<String, List<ErrorProneEntry>> e : errors.entrySet()) {
+                String errorType = e.getKey();
+                List<ErrorProneEntry> errorsOfType = e.getValue();
+                if (errorsOfType.isEmpty()) {
+                    continue;
+                }
+
+                for (ErrorProneEntry error : e.getValue()) {
+                    wrt.print("| ");
+                    wrt.print("[`" + errorType + "`](" + error.url + ")");
+                    String tags = ErrorProneUtils.getTags(errorType);
```

Comment:
optional: you can pull this out of the loop, together with computing either "" or `" " + escapeMd(tags)` --> just unconditionally append the resulting string here

Reply:
Done